### PR TITLE
Implement agent

### PR DESCRIPTION
**Pull request by f0m41h4u7**
implement agent that retrieves WG config from manager and applies it on host

### CODE REVIEW

File: tests/Dockerfile_agent [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+RUN KVER=`uname -r`
+RUN apt-get install -y linux-headers-$KVER && apt-get install -y --no-install-recommends wireguard-dkms bc
+
+ENV GO111MODULE=on
+ENV GOROOT=/usr/local/go
+ENV GOPATH=/home/go
+ENV PATH=$PATH:/usr/local/go/bin
+
+WORKDIR /cloud-manager
```
Comment (n0str) [inline tests/Dockerfile_agent:23]:
```suggestion
WORKDIR /cloud-agent
```

File: tests/test.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (n0str) [inline tests/test.sh:1]: some mess with directories. Should be `docker-compose -f docker-compose-test.yaml` in all cases

File: tests/docker-compose-test.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      - test
+
+  manager:
+    container_name: manager
+    hostname: manager
+    build:
+      context: ../
+      dockerfile: ./tests/Dockerfile_manager
+    volumes:
+      - ${PWD}/tests/config.yaml:/cloud-manager/examples/config.yaml
```
Comment (n0str) [inline tests/docker-compose-test.yaml:21]:
```suggestion
      - ${PWD}/config.yaml:/cloud-manager/examples/config.yaml
```

**Review by n0str — CHANGES_REQUESTED** [review]

File: tests/test.sh (hunk 2 above)
Comment (f0m41h4u7) [reply tests/test.sh]: actually this script was called from Makefile, which is outside `tests` directory. 
i've moved this script's logic to Makefile

File: tests/docker-compose-test.yaml (hunk 3 above)
Comment (f0m41h4u7) [reply tests/docker-compose-test.yaml:21]: same reason. `docker-compose up` is called from Makefile, and the path to `config.yaml` is set here regarding to Makefile location

**Review by n0str — APPROVED** [review]

File: cmd/agent/main.go [hunk 4]
```diff
@@ -1,5 +1,93 @@
...
+	}
 
+	// Query new config from Manager periodically
+	ticker := time.NewTicker(interval)
+	stop := false
+	for !stop {
+		select {
+		case <-done:
+			signal.Stop(done)
+			stop = true
```
Comment (kabachook) [inline cmd/agent/main.go:83]: graceful shutdown
`wg-quick down wg0`

**Review by kabachook — CHANGES_REQUESTED** [review]

File: cmd/agent/main.go (hunk 4 above)
Comment (kabachook) [reply cmd/agent/main.go:83]: рассмотреть случай, когда при запуске интерфейс уже поднят и `wg-quick up` выдает ошибку (exit code 1)

**Review by kabachook — APPROVED** [review]

**Pull request merged**